### PR TITLE
Add typescript logo

### PR DESCRIPTION
**Pull request by styfle**
Add the TypeScript logo. I'm not sure if this will look okay so let me know if it needs to be changed @amio

### CODE REVIEW

File: icons/typescript.svg [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (amio) [inline icons/typescript.svg]:
```suggestion
```

**Comment by amio** [comment]
It looks nice, thanks! @styfle

**Comment by styfle** [comment]
Hmm, it looks a little too large 🤔 

https://badgen.net/badge/-/typescript?icon=typescript&label

img

Does the padding need to be adjusted in the svg or in badgen.net?

**Comment by amio** [comment]
In the svg. Got the same feeling too :D

How about this?

image

```
<svg xmlns="http://www.w3.org/2000/svg" version="1" viewBox="-35 -35 700 700">
  <path fill="#FFF" d="M0 315v315h630V0H0v315zm508-25c16 4 28 11 39 23 6 6 15 17 15 20 0 1-27 19-44 29 0 1-3-2-5-6-8-12-17-17-30-18-19-1-31 9-31 26 0 5 1 7 3 11 4 9 12 14 36 25 45 19 64 32 76 50 14 21 17 53 8 77-10 26-35 44-70 50-11 2-36 2-48 0-25-5-49-17-64-34-6-6-17-23-17-24l6-4 24-13 18-11 4 6c5 8 16 19 23 23 21 10 48 9 62-3 6-6 8-11 8-19s-1-11-5-16c-5-7-15-13-43-25-33-14-47-23-59-37-8-8-15-21-18-31-2-9-3-32-1-41 7-31 31-53 65-59 11-3 37-2 48 1zm-148 26v26h-82v233h-58V342h-82v-25l1-26 110-1h111v26z"/>
</svg>
```

**Comment by amio** [comment]
Maybe this one

image
```
<svg xmlns="http://www.w3.org/2000/svg" version="1" viewBox="-20 -20 670 670">
  <path fill="#FFF" d="M0 315v315h630V0H0v315zm508-25c16 4 28 11 39 23 6 6 15 17 15 20 0 1-27 19-44 29 0 1-3-2-5-6-8-12-17-17-30-18-19-1-31 9-31 26 0 5 1 7 3 11 4 9 12 14 36 25 45 19 64 32 76 50 14 21 17 53 8 77-10 26-35 44-70 50-11 2-36 2-48 0-25-5-49-17-64-34-6-6-17-23-17-24l6-4 24-13 18-11 4 6c5 8 16 19 23 23 21 10 48 9 62-3 6-6 8-11 8-19s-1-11-5-16c-5-7-15-13-43-25-33-14-47-23-59-37-8-8-15-21-18-31-2-9-3-32-1-41 7-31 31-53 65-59 11-3 37-2 48 1zm-148 26v26h-82v233h-58V342h-82v-25l1-26 110-1h111v26z"/>
</svg>
```

You can experiment markups on https://badgen-icon-preview.now.sh

**Comment by styfle** [comment]
Oh nice, I missed the preview. That's helpful!

I think the smallest one is best.

Also, it would be really cool if there was a way to specify the color of the left side and the color of the icon. I could imaging a blue TS with a white background.

Logo: https://cdn.jsdelivr.net/gh/remojansen/logo.ts@master/ts.svg

Also checkout the Jest badge: [jest](https://github.com/facebook/jest)

**Comment by amio** [comment]
A white bg would fade in to page bg, blur the edge of a badge. I would make it darker if I do the jest badge :P

There's a lot demands on changing styles on left side, I'm sure it can goes to `badgen` library (will do it later). While I'm not sure if we should expose this on badgen.net, for:
- less configuration
- consistent style

Hmm, this discussion gives me another idea, maybe we could add a `style=bright` or something on badgen.net, to have a style set for dark theme that has become increasingly popular nowadays.

**Comment by styfle** [comment]
@amio So maybe `style=light` would be a lighter color on the left side and `style=dark` would be a darker color (current color) on the left side?

I like it! 💯

**Comment by amio** [comment]
Yea, that's the idea. `light` vs `dark` sounds great! 🏂

**Pull request merged**